src/SingleProject/Resizetizer/src/GenerateSplashStoryboard.cs

```diff
@@ -9,24 +11,80 @@ namespace Microsoft.Maui.Resizetizer
 	/// <summary>
 	/// Generates the MauiSplash.storyboard file for iOS splash screens
 	/// </summary>
-	public class GenerateSplashStoryboard : Task
+	public class GenerateSplashStoryboard : Task, ILogger
 	{
 		[Required]
-		public string OutputFile { get; set; }
+		public string IntermediateOutputPath { get; set; }
 
 		[Required]
 		public ITaskItem[] MauiSplashScreen { get; set; }
 
+		public string InputsFile { get; set; }
+
 		public override bool Execute()
 		{
-			var splash = MauiSplashScreen[0];
+			Log.LogMessage(MessageImportance.Low, $"Splash Screen: Intermediate Path " + IntermediateOutputPath);
+
+			var splash = MauiSplashScreen?.FirstOrDefault();
+			if (splash is null)
+			{
+				Log.LogMessage(MessageImportance.Low, $"Splash Screen: No images found.");
+				return true;
+			}
+
+			try
+			{
+				var info = ResizeImageInfo.Parse(splash);
+
+				var resizer = new Resizer(info, IntermediateOutputPath, this);
+
+				WriteImages(resizer);
+				WriteStoryboard(resizer);
 
-			var info = ResizeImageInfo.Parse(splash);
+				return !Log.HasLoggedErrors;
+			}
+			catch (Exception ex)
+			{
+				Log.LogError(null, "MAUI0000", null, null, 0, 0, 0, 0, ex.ToString());
+				return false;
+			}
+		}
 
-			var outputFileName = info.OutputName;
-			var image = outputFileName + ".png";
+		private void WriteImages(Resizer resizer)
+		{
+			if (resizer.Info.Resize)
+			{
+				foreach (var dpi in DpiPath.Ios.Image)
+				{
+					Log.LogMessage(MessageImportance.Low, $"Splash Screen Resize: " + dpi);
+					resizer.Resize(dpi, InputsFile);
+				}
+			}
+			else
+			{
+				var dpi = DpiPath.Ios.Original;
 
-			var color = info.Color ?? SKColors.White;
+				Log.LogMessage(MessageImportance.Low, $"Splash Screen Copy: " + dpi);
+				resizer.CopyFile(dpi, InputsFile);
+			}
+		}
+
+		private void WriteStoryboard(Resizer resizer)
+		{
+			Directory.CreateDirectory(IntermediateOutputPath);
+			var storyboardFile = Path.Combine(IntermediateOutputPath, "MauiSplash.storyboard");
+
+			Log.LogMessage(MessageImportance.Low, $"Splash Screen Storyboard: " + storyboardFile);
+
+			DpiPath dpi;
+			if (resizer.Info.Resize)
+				dpi = DpiPath.Ios.Image[0];
```

Comment:
Does this need to do anything if the array is empty, or contains more than one?

Reply:
This is a hardcoded set of the pre-defined image sizes. The list contains the sizes of the image scales, with `[0]` being the "base" size. There is more than one and if there are none then maui is horribly broken. I have some unit tests around this to make sure that there is always the valid scales.